libs/infra-nest-server/src/lib/bootstrap.ts

```diff
@@ -72,7 +73,10 @@ export const createApp = async (options: RunServerOptions) => {
 
   // Enable validation of request DTOs globally.
   app.useGlobalPipes(
-    new ValidationPipe({ whitelist: true, forbidNonWhitelisted: true }),
```

Comment:
whitelisting does not interact well with GraphQL:
```typescript
import { Field, InputType } from '@nestjs/graphql'
@InputType()
export class DelegationInput {
  @Field((_) => String)
  name!: string
  
  @Field((_) => String)
  toNationalId!: string
}
```
this for instance won't work, since it'll be stripped in the API even though the client provides the correct input, so a 400 bad request will be thrown.

By whitelisting, class-validator enforces GraphQL to decorate every field with any class-validator decorator in order to pick it up, just for the sake of it... so the previous example needs to be changed to:
```typescript
import { Field, InputType } from '@nestjs/graphql'
import { Allow } from 'class-validator'
@InputType()
export class DelegationInput {
  @Allow()
  @Field((_) => String)
  name!: string
  
  @Allow()
  @Field((_) => String)
  toNationalId!: string
}
```
in order to prevent nestjs's validator from stripping it.

Reply:
So GraphQL and Class Validator both solve this same thing:

* Validate data types and structure.
* Trim extraneous inputs.

Additionally, the class validator has many ways to validate input data, including string length, regex, etc. Arguably all fields should have some class validator decorator, as it's important to validate all user input. But that's probably the only reason to have `whitelist: true` in our GraphQL project, to force the developer to give this some thought.

Your call. We can disable whitelist in the graphql API, but I want to keep it in the REST APIs.

Reply:
Actually, probably most of the graphql DTOs are directly transferred to internal services. Arguably proper validation should live there instead, so we don't have out of sync validation in our backend layers.

So let's disable the whitelist, but make sure that all user input is validated.

Reply:
Yeah I agree, opting out would be a better approach.
Implemented in: https://github.com/island-is/island.is/pull/4063/commits/7b8c4f225b986526700936990544fa316141e140